AnKi/Gr/Vulkan/GrManagerImpl.cpp

```diff
@@ -1012,16 +1012,43 @@ Error GrManagerImpl::initDevice(const GrManagerInitInfo& init)
 		if(!m_fragmentShadingRateFeatures.attachmentFragmentShadingRate
 		   || !m_fragmentShadingRateFeatures.pipelineFragmentShadingRate)
 		{
-			ANKI_VK_LOGE(VK_KHR_FRAGMENT_SHADING_RATE_EXTENSION_NAME
+			ANKI_VK_LOGW(VK_KHR_FRAGMENT_SHADING_RATE_EXTENSION_NAME
 						 " doesn't support attachment and/or pipeline rates");
-			return Error::FUNCTION_FAILED;
+			m_capabilities.m_vrs = false;
 		}
 
 		// Disable some things
 		m_fragmentShadingRateFeatures.primitiveFragmentShadingRate = false;
 
-		m_fragmentShadingRateFeatures.pNext = const_cast<void*>(ci.pNext);
-		ci.pNext = &m_fragmentShadingRateFeatures;
+		VkPhysicalDeviceFragmentShadingRatePropertiesKHR fragmentShadingRateProperties = {};
+		fragmentShadingRateProperties.sType = VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_FRAGMENT_SHADING_RATE_PROPERTIES_KHR;
+
+		VkPhysicalDeviceProperties2 properties = {};
+		properties.sType = VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_PROPERTIES_2;
+		properties.pNext = &fragmentShadingRateProperties;
+		vkGetPhysicalDeviceProperties2(m_physicalDevice, &properties);
+
+		if(fragmentShadingRateProperties.minFragmentShadingRateAttachmentTexelSize.width > 16
```

Comment:
I guess that will disable VRS on certain platforms that go up to 32x32. I guess this check should be if it doesn't support 16x16 or it doesn't support 8x8.

Reply:
Going up to 32x32 is fine as long as 16x16 is still supported.
To check if 8x8 or 16x16 is supported is:
`(minFragmentShadingRateAttachmentTexelSize <= 16 <= maxFragmentShadingRateAttachmentTexelSize) || (minFragmentShadingRateAttachmentTexelSize <= 8 <= maxFragmentShadingRateAttachmentTexelSize)`
which can be simplified to:
`minFragmentShadingRateAttachmentTexelSize <= 16 && maxFragmentShadingRateAttachmentTexelSize >= 8`
Inverse is
`minFragmentShadingRateAttachmentTexelSize > 16 || maxFragmentShadingRateAttachmentTexelSize < 8`

Reply:
nvm. I missed the "min" in minFragmentShadingRateAttachmentTexelSize